src/components/FormElements/AssetPicker.tsx

```diff
@@ -5,29 +5,40 @@ import { ExpandBox } from "@/components/ExpandBox";
 import { RadioPicker } from "@/components/RadioPicker";
 import { PubKeyPicker } from "@/components/FormElements/PubKeyPicker";
 import { LiquidityPoolShares } from "@/components/FormElements/LiquidityPoolShares";
+import { TextPicker } from "@/components/FormElements/TextPicker";
 
 import {
   AssetError,
   AssetObject,
   AssetObjectValue,
   AssetPoolShareError,
   AssetPoolShareObjectValue,
+  AssetSinglePoolShareValue,
 } from "@/types/types";
 
 type AssetPickerProps = {
   id: string;
   label: string;
   labelSuffix?: string | React.ReactNode;
-  value: AssetObjectValue | AssetPoolShareObjectValue | undefined;
+  value:
+    | AssetObjectValue
+    | AssetPoolShareObjectValue
+    | AssetSinglePoolShareValue
+    | undefined;
   error: AssetError | AssetPoolShareError | undefined;
   note?: React.ReactNode;
   onChange: (
-    asset: AssetObjectValue | AssetPoolShareObjectValue | undefined,
+    asset:
+      | AssetObjectValue
+      | AssetPoolShareObjectValue
+      | AssetSinglePoolShareValue
+      | undefined,
   ) => void;
   assetInput: "issued" | "alphanumeric";
   fitContent?: boolean;
   includeNative?: boolean;
   includeLiquidityPoolShares?: boolean;
+  includePoolShare?: boolean;
```

Comment:
Maybe rename it something like this to differentiate it from the above `includeLiquidityPoolShares`?
```suggestion
  includeSingleLiquidityPoolShare?: boolean;
```